### ISSUE TITLE
[Bug]: Could not find expected browser locally

### ISSUE THREAD

**Issue by ImRodry** [puppeteer/puppeteer#9119]
### Bug description

I started noticing this issue as soon as I updated to v19. I didn’t change any of my code as the list of breaking changes didn’t make it seem like I had to but if that’s the issue please let me know. The issue occurs as soon as I called `puppeteer.launch()`. 

Steps to reproduce the problem:

1. Install v19
2. Launch a new chrome browser using `puppeteer.launch({ args: ["args: ["--no-sandbox"], headless: true })`
3. Observe the error

### Puppeteer version

19.0.0

### Node.js version

18.10.0

### npm version

Yarn 1.22.19

### What operating system are you seeing the problem on?

Windows

### Relevant log output

```shell
/app/node_modules/puppeteer-core/lib/cjs/puppeteer/node/ChromeLauncher.js:69
                  throw new Error(missingText);
                        ^
  Error: Could not find expected browser (chrome) locally. Run `npm install` to download the correct Chromium revision (1045629).
      at ChromeLauncher.launch (/app/node_modules/puppeteer-core/src/node/ChromeLauncher.ts:118:15)
```

**Comment by mlahp7** [comment]
Same error for me, going to revert upgrade for now.

`Could not find expected browser (chrome) locally. Run `npm install` to download the correct Chromium revision (1045629)`

**Comment by jrandolf** [comment]
Yarn typically has issues due to caching. Could you clean yarn's cache and try again? It would also make sense to "git clean -X" to remove any gitignored files and try a clean install. If this does not work, is your home directory accessible? Since the browser binaries are now placed in a home directory .cache folder, puppeteer needs permission to create directories there. To verify if this is the case, set PUPPETEER_CACHE_DIR to your current working directory and see if puppeteer starts downloading there.

**Comment by ImRodry** [comment]
@jrandolf I'm getting this error on my host which performs a clean install on every push, so I doubt that would change anything. As for the home directory, I probably don't have access to that and it would be very dumb to require that as that would fall outside of the directory of the app which most hosts don't let you access, so that should be changed if that's the case.

**Comment by jrandolf** [comment]
In general, a cloud host provider will create a user that is specific to the app or your account. Sometimes a home directory is created, other times not, but I wouldn't say it's dumb since a home directory really means "the base directory that we allow the user to have". In any case, we plan to use 'XDG_CACHE_HOME' in the future which may be defined on your provider, but I'd recommend using PUPPETEER_CACHE_DIR as this will give you the most control regarding where you download your browser.

**Comment by ImRodry** [comment]
What I'm saying is that it doesn't make sense to download the browser outside of the app's directory by default. I understand I can control it like that but the default should remain within the app and not require us to set an environment variable for it to work which wasn't even documented or alerted in the changelog

**Comment by jrandolf** [comment]
This was documented:

> use ~/.cache/puppeteer for browser downloads

If you inspect the PR, we also document how to work with it if that directory is problematic.

In any case, there are a few solutions that we are discussing regarding this issue (we know using environment variables is a bad solution). Most likely we will have something like .puppeteerrc, but for now, we recommend using the environment variable. Does this solve your issue?

**Comment by ImRodry** [comment]
Im not sure because I added the variable and the issue was fixed but then I removed it and it was still working but I’ll keep the environment variable just in case. I really think you should change the default back to what it was before so that it doesn’t require a folder outside of the app’s directory.

**Comment by ImRodry** [comment]
@jrandolf seems like that didn’t fix it after all. Could you reopen this?
I set the env var to the same as my WORKDIR in my dockerfile, any tips?

**Comment by jrandolf** [comment]
As you are using a Dockerfile, our support for setup is limited there. We ask that you use the docker image we provide as a base/inspiration since it's tested in our CI. See [pptr.dev/guides/docker](https://pptr.dev/guides/docker/).

**Comment by dmcgougan** [comment]
This broke our jenkins build server

**Comment by ImRodry** [comment]
@jrandolf the dockerfile is not the cause of the issue. I'm no longer using one and the issue is still happening and clearly it is for other people too. Please reopen this

**Comment by mo** [comment]
I got this problem as well and I'm not using cloud function, not lambda and not even docker. I also use standard "npm" (i.e. not yarn or pnpm). I started "node" and ran "os.homedir().toString()" in the REPL and it finds my homedir and my homedir is readable. I'm using ubuntu 22.04 on this computer. I looked in ~/.cache/puppeteer and there was a copy of chrome in that directory but the app still crashed with the message "Could not find expected browser locally" when it ran the puppeteer.launch() code.

However, then I ran:

```rm -rf ~/.cache/puppeteer/ ; rm -rf node_modules/```

....and then "npm install" and after this everything worked again.
